src/Server/Auth.fs

```diff
@@ -10,31 +10,14 @@ let login next (ctx: HttpContext) = task {
     let! login = FableJson.getJsonFromCtx<Domain.Login> ctx
 
     try
-        if (login.UserName <> "test" || login.Password <> "test") && 
+        if (login.UserName <> "test" || login.Password <> "test") &&
            (login.UserName <> "test2" || login.Password <> "test2") then
             return! failwithf "Could not authenticate %s" login.UserName
-        let user : ServerTypes.UserRights = { UserName = login.UserName }
-        let userData :Domain.UserData = { UserName = login.UserName; Token = JsonWebToken.encode user }
+        let token = JsonWebToken.generateToken login.UserName
+        let userData :Domain.UserData = { UserName = login.UserName; Token = token }
         return! FableJson.serialize userData next ctx
     with
-    | _ -> 
+    | _ ->
         return! UNAUTHORIZED "Bearer" "" (sprintf "User '%s' can't be logged in." login.UserName) next ctx
 }
 
-/// Invokes a function that produces the output for a web part if the HttpContext
-/// contains a valid auth token. Use to authorise the expressions in your web part
-/// code (e.g. WishList.getWishList).
-let useToken next (ctx: HttpContext) f = task {
-    match ctx.Request.Headers.TryGetValue "Authorization" with
-    | true, accesstoken -> 
-        match Seq.tryHead accesstoken with 
-        | Some t when t.StartsWith "Bearer " ->
-            let jwt = t.Replace("Bearer ","")
-            match JsonWebToken.isValid jwt with
-            | None -> return! FORBIDDEN "Accessing this API is not allowed" next ctx
-            | Some token -> return! f token
-        | _ ->
-           return! BAD_REQUEST "Request doesn't contain a JSON Web Token" next ctx
-    | _ -> 
-        return! BAD_REQUEST "Request doesn't contain a JSON Web Token" next ctx
-}
```

Comment:
@Krzysztof-Cieslak I assume that this whole block is rendered obsolete by Saturn?

Reply:
Yes, indeed. Saturn default JWT helpers are using authorization implementation provided by ASP.NET which means that all handling of cases where token is not provided, token is not valid etc is already provided. 

It changes a bit a way how you use token - in old implementation `useToken` was "higher order handler" which fed token to out actions. In the new one, ASP.NET is populating `HttpContext.User` field. While it includes some ASP.NET magic it makes the authentication system totally pluggable - you can easily change authentication to Cookies, or external OAuth or anything else that's supported by ASP.NET and framework will still populate same `HttpContext.User` field with authentication information.

Reply:
So it's not really made obsolete by Saturn - it's just side effect of Saturn using ASP.NET mechanisms for authentication helpers. 

Same would happen if we moved to ASP.NET authentication without using Saturn, and just Giraffe's `challenge` and `requiresAuthentication` handlers